### PR TITLE
Panos mgtconfig

### PR DESCRIPTION
**Pull request by ivanbojer**
##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
panos_mgtconfig

##### ANSIBLE VERSION

```
ansible 2.1.0.0 
config file = 
configured module search path = Default w/o overrides
```

##### SUMMARY
New panos module that allows for rudimentary management interface configuration.

### CODE REVIEW

File: lib/ansible/modules/network/panos/panos_mgtconfig.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  panos_mgtconfig:
+    ip_address: "192.168.1.1"
+    password: "admin"
+    dns_server_primary: "1.1.1.1"
+    dns_server_secondary: "1.1.1.2"
+    panorama_primary: "1.1.1.3"
+    panorama_secondary: "1.1.1.4"
+'''
+
+RETURN = '''
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_mgtconfig.py]: doc string doesn't match actual return value

File: lib/ansible/modules/network/panos/panos_mgtconfig.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        panorama_primary=dict(),
+        panorama_secondary=dict(),
+        commit=dict(type='bool', default=True)
+    )
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=False)
+    if not HAS_LIB:
+        module.fail_json(msg='pan-python is required for this module')
+
+    ip_address = module.params["ip_address"]
+    if not ip_address:
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_mgtconfig.py]: not needed

File: lib/ansible/modules/network/panos/panos_mgtconfig.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+        module.fail_json(msg="ip_address should be specified")
+    password = module.params["password"]
+    if not password:
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_mgtconfig.py]: not needed

File: lib/ansible/modules/network/panos/panos_mgtconfig.py [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+            changed |= set_dns_server(xapi, dns_server_secondary, primary=False)
+        if panorama_primary is not None:
+            changed |= set_panorama_server(xapi, panorama_primary, primary=True)
+        if panorama_secondary is not None:
+            changed |= set_panorama_server(xapi, panorama_secondary, primary=False)
+
+        if changed and commit:
+            xapi.commit(cmd="<commit></commit>", sync=True, interval=1)
+    except PanXapiError:
+        import sys
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_mgtconfig.py]: use `get_exception()` instead

File: lib/ansible/modules/network/panos/panos_mgtconfig.py [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+    ip_address = module.params["ip_address"]
+    if not ip_address:
+        module.fail_json(msg="ip_address should be specified")
+    password = module.params["password"]
+    if not password:
+        module.fail_json(msg="password is required")
+    username = module.params['username']
+
+    xapi = pan.xapi.PanXapi(
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_mgtconfig.py:179]: shared function?

**Review by privateip — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/panos/panos_mgtconfig.py (hunk 1 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_mgtconfig.py]: mimicked wakeonlan.py given there is no return var

File: lib/ansible/modules/network/panos/panos_mgtconfig.py (hunk 2 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_mgtconfig.py]: ok

File: lib/ansible/modules/network/panos/panos_mgtconfig.py (hunk 3 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_mgtconfig.py]: ok

File: lib/ansible/modules/network/panos/panos_mgtconfig.py (hunk 5 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_mgtconfig.py:179]: nope, just a class instantiation

File: lib/ansible/modules/network/panos/panos_mgtconfig.py (hunk 4 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_mgtconfig.py]: ok

**Comment by ivanbojer** [comment]
ready_for_review

**Pull request merged**